power_grid_model_c/power_grid_model/include/power_grid_model/component/power_sensor.hpp

```diff
@@ -59,19 +59,19 @@ class GenericPowerSensor : public Sensor {
 
     // virtual function getter for sym and asym param
     // override them in real sensors function
-    virtual PowerSensorCalcParam<true> sym_calc_param() const = 0;
-    virtual PowerSensorCalcParam<false> asym_calc_param() const = 0;
+    virtual PowerSensorCalcParam<symmetric_t> sym_calc_param() const = 0;
+    virtual PowerSensorCalcParam<asymmetric_t> asym_calc_param() const = 0;
 
-    virtual PowerSensorOutput<true> get_sym_output(ComplexValue<true> const& s) const = 0;
-    virtual PowerSensorOutput<false> get_asym_output(ComplexValue<false> const& s) const = 0;
+    virtual PowerSensorOutput<symmetric_t> get_sym_output(ComplexValue<symmetric_t> const& s) const = 0;
+    virtual PowerSensorOutput<asymmetric_t> get_asym_output(ComplexValue<asymmetric_t> const& s) const = 0;
 };
 
-template <bool sym> class PowerSensor : public GenericPowerSensor {
+template <symmetry_tag sym> class PowerSensor : public GenericPowerSensor {
```

Comment:
idem as for load_gen.hpp: we may want to add separate tags here. that would fix the ambiguity between (a)symmetric calculation type and (a)symmetric component type. I would propose to do that change in `base.hpp`

Reply:
See the comment above. Do not create additional tag types, but use proper template argument name.